src/js/components/dialog.js

```diff
@@ -406,6 +406,9 @@ export default class Dialog extends Component {
           'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="-1"]'
         )
 
+        this.focusableChildElements = Array.from(this.focusableChildElements)
+        this.focusableChildElements = this.focusableChildElements.filter(el => el.checkVisibility()) // Exclude hidden elements
```

Comment:
This is one of those gray areas where we may need to look at browser support/adoption a bit more closely. the `Element.checkVisibility()` while it technically is available in all modern/evergreen browsers, it is _super_ new. My biggest concern here is that it was only release ind Safari (macOS and iOS) back in May 2024. Knowing anecdotally that a lot of iOS and macOS users tend to not always update right away because it take manual action to do so (unlike Chrome, Firefox, etc.), we may want to wait a bit before we start using this method in production. Especially since we have such high traffic on our sites from iOS Safari.

We'll want to go with something simpler like checking the clientHeight, etc.

Reply:
@scottanthonymurray Sorry, meant to share this link to the browser stats https://caniuse.com/mdn-api_element_checkvisibility